### ISSUE TITLE
New-AzureRmVmss show incorect message HealthProbe resource missing

### ISSUE THREAD

**Issue by DanielHWe** [Azure/azure-powershell#6413]
### Description

I create first a Network Load Balancer include HealthProbe etc. and then on create the according Vm Scale Set with New-AzureRmVmss, I got the message that the HealthProbe is missing. But by checking on the azure portel the HealthProbe was correctly create within the Load Balancer resource.

### Script/Steps for Reproduction

```powershell
# Set required Variables
$lbName = "NRP-LB"
$loc = "<My Region>"
$currentrg = "<My rg name>"
$vnetname = "<My Vnet Name>"
$vnetrg = "<Name of RG conteining my vnet>"

$vmPassword = ConvertTo-SecureString $adminPassword -AsPlainText -Force
$vmCred = New-Object System.Management.Automation.PSCredential($adminUsername, $vmPassword)

# Set Network
$vnet= Get-AzureRmVirtualNetwork -Name $vnetname -ResourceGroupName $vnetrg
$subnetMain = $vnet.Subnets[2]
$backendSubnetName = $subnetMain.Name


#Deploy Loadbalancer
$frontendIP = New-AzureRmLoadBalancerFrontendIpConfig `
     -Name LB-Frontend `
     -SubnetId $subnetMain.Id `
     -ErrorAction Stop 

$beaddresspool= New-AzureRmLoadBalancerBackendAddressPoolConfig `
     -Name "LB-backend" `
     -ErrorAction Stop

$inboundNATRule1= New-AzureRmLoadBalancerInboundNatRuleConfig `
     -Name "RDP" `
     -FrontendIpConfiguration $frontendIP `
     -Protocol TCP `
     -FrontendPort 3389 `
     -BackendPort 3389 `
     -ErrorAction Stop

$healthProbe = New-AzureRmLoadBalancerProbeConfig `
     -Name "HealthProbe" `
     -RequestPath "Default.htm" `
     -Protocol http `
     -Port 80 `
     -IntervalInSeconds 15 `
     -ProbeCount 2 `
     -ErrorAction Stop

$lbrule = New-AzureRmLoadBalancerRuleConfig `
     -Name "HTTP" `
     -FrontendIpConfiguration $frontendIP `
     -BackendAddressPool $beAddressPool `
     -Probe $healthProbe `
     -Protocol Tcp `
     -FrontendPort 80 `
     -BackendPort 80

$nrplb = New-AzureRmLoadBalancer `
     -ResourceGroupName $currentrg `
     -Name $lbName `
     -Location $loc `
     -FrontendIpConfiguration $frontendIP `
     -InboundNatRule $inboundNATRule1 `
     -LoadBalancingRule $lbrule `
     -BackendAddressPool $beAddressPool `
     -Probe $healthProbe 

#Deploy VM Scale Set ==> Error
New-AzureRmVmss `
  -Credential $vmCred `
  -ResourceGroupName $currentrg `
  -Location $loc `
  -VMScaleSetName $vmssName `
  -PublicIpAddressName $frontendIP.Name `
  -LoadBalancerName  $lbName `
  -UpgradePolicy "Automatic" `
  -ErrorAction Stop `
  -VirtualNetworkName $vnetname `
  -SubnetName $backendSubnetName `
```

### Module Version

6.2.1

### Environment Data

Name                           Value                                                                                                            
----                           -----                                                                                                            
PSVersion                      5.1.16299.431                                                                                                    
PSEdition                      Desktop                                                                                                          
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                          
BuildVersion                   10.0.16299.431                                                                                                   
CLRVersion                     4.0.30319.42000                                                                                                  
WSManStackVersion              3.0                                                                                                              
PSRemotingProtocolVersion      2.3                                                                                                              
SerializationVersion           1.1.0.1    

### Debug Output

```
WARNUNG: The output object type of this cmdlet will be modified in a future release.

New-AzureRmVmss : Resource /subscriptions/XXX/resourceGroups/XXX/providers/Microsoft.Network/loadBal
ancers/NRP-LB/loadBalancingRules/HTTP referenced by resource /subscriptions/XXX/resourceGroups/XXX/p
roviders/Microsoft.Network/loadBalancers/NRP-LB/probes/HealthProbe was not found. Please make sure that the referenced resource exists, and 
that both resources are in the same region.
In D:\scripts\create_loadBalancer_and_vmss.ps1:228 Zeichen:1
+ New-AzureRmVmss `
+ ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzureRmVmss], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmss
```

**Comment by markcowl** [comment]
@praries880 @hyonholee Can you take a look?

**Comment by praries880** [comment]
@DanielHWe  I have a local repro as well.

@hyonholee 
Here are the steps i ran :
```powershell
$publicip = New-AzureRmPublicIpAddress -ResourceGroupName "parytest1" -Name "MyPublicIp" -Location "East US" -AllocationMethod "Dynamic"
$frontend = New-AzureRmLoadBalancerFrontendIpConfig -Name "MyFrontEnd" -PublicIpAddress $publicip
$backendAddressPool = New-AzureRmLoadBalancerBackendAddressPoolConfig -Name "MyBackendAddPoolConfig02"
$probe = New-AzureRmLoadBalancerProbeConfig -Name "MyProbe" -Protocol "http" -Port 80 -IntervalInSeconds 15 -ProbeCount 2 -RequestPath "healthcheck.aspx"
$inboundNatRule1 = New-AzureRmLoadBalancerInboundNatRuleConfig -Name "MyinboundNatRule1" -FrontendIPConfiguration $frontend -Protocol "Tcp" -FrontendPort 3389 -BackendPort 3389 -IdleTimeoutInMinutes 15 -EnableFloatingIP
$inboundNatRule2 = New-AzureRmLoadBalancerInboundNatRuleConfig -Name "MyinboundNatRule2" -FrontendIPConfiguration $frontend -Protocol "Tcp" -FrontendPort 3391 -BackendPort 3392
$lbrule = New-AzureRmLoadBalancerRuleConfig -Name "MyLBruleName" -FrontendIPConfiguration $frontend -BackendAddressPool $backendAddressPool -Probe $probe -Protocol "Tcp" -FrontendPort 80 -BackendPort 80 -IdleTimeoutInMinutes 15 -EnableFloatingIP -LoadDistribution SourceIP
$lb = New-AzureRmLoadBalancer -Name "MyLoadBalancer" -ResourceGroupName "parytest1" -Location "East US" -FrontendIpConfiguration $frontend -BackendAddressPool $backendAddressPool -Probe $probe -InboundNatRule $inboundNatRule1,$inboundNatRule2 -LoadBalancingRule $lbrule

New-AzureRmVmss  -ResourceGroupName parytest1   -Location "East US"   -VMScaleSetName "paryvmss1"
 -PublicIpAddressName $frontendIP.Name   -LoadBalancerName  "MyLoadBalancer"   -UpgradePolicy "Automatic"   -ErrorAction
 Stop

cmdlet New-AzureRmVmss at command pipeline position 1
Supply values for the following parameters:
Credential
New-AzureRmVmss : Resource /subscriptions/<redacted>/resourceGroups/parytest1/providers/Micro
soft.Network/loadBalancers/MyLoadBalancer/loadBalancingRules/MyLBruleName referenced by resource /subscriptions/<redacted>/resourceGroups/parytest1/providers/Microsoft.Network/loadBalancers/MyLoadBalancer/probes/
MyProbe was not found. Please make sure that the referenced resource exists, and that both resources are in the same
region.
At line:1 char:1
+ New-AzureRmVmss  -ResourceGroupName parytest1   -Location "East US"   ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzureRmVmss], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmss

```

The LB Rule does exist, as can be seen here:
```powershell
$lb = Get-AzureRmLoadBalancer -Name "MyLoadBalancer" -ResourceGroupName "parytest1"

PS C:\Users\paresin> $lb | select Name, ResourceGroupName

Name           ResourceGroupName
----           -----------------
MyLoadBalancer parytest1


PS C:\Users\paresin> $lb.Probes | select Name, RequestPath, LoadBalancingRules

Name    RequestPath      LoadBalancingRules
----    -----------      ------------------
MyProbe healthcheck.aspx {/subscriptions/<redacted>/resourceGroups/parytest1/providers/Mic...


PS C:\Users\paresin> $lb.LoadBalancingRules | select Name, ProvisioningState

Name         ProvisioningState
----         -----------------
MyLBruleName Succeeded
```

The error seems to be coming from the service itself : 
```
RequestId      : 303c2d29-f001-4bf4-8ab1-c6bf9dd89672
Message        : Resource /subscriptions/<redacted>/resourceGroups/parytest1/providers/Micros
                 oft.Network/loadBalancers/MyLoadBalancer/loadBalancingRules/MyLBruleName referenced by resource /subsc
                 riptions/<redacted>/resourceGroups/parytest1/providers/Microsoft.Network/loa
                 dBalancers/MyLoadBalancer/probes/MyProbe was not found. Please make sure that the referenced resource
                 exists, and that both resources are in the same region.
ServerMessage  : InvalidResourceReference: Resource /subscriptions/<redacted>/resourceGroups/
                 parytest1/providers/Microsoft.Network/loadBalancers/MyLoadBalancer/loadBalancingRules/MyLBruleName
                 referenced by resource /subscriptions/<redacted>/resourceGroups/parytest1/pr
                 oviders/Microsoft.Network/loadBalancers/MyLoadBalancer/probes/MyProbe was not found. Please make sure
                 that the referenced resource exists, and that both resources are in the same region.
                 (System.Collections.Generic.List`1[Microsoft.Rest.Azure.CloudError])
ServerResponse : {BadRequest}
RequestMessage : {PUT https://management.azure.com/subscriptions/<redacted>/resourceGroups/pa
                 rytest1/providers/Microsoft.Network/loadBalancers/MyLoadBalancer?api-version=2017-10-01}
InvocationInfo : {New-AzureRmVmss}
Line           : New-AzureRmVmss  -ResourceGroupName parytest1   -Location "East US"   -VMScaleSetName "paryvmss1"
                 -PublicIpAddressName $frontendIP.Name   -LoadBalancerName  "MyLoadBalancer"   -UpgradePolicy
                 "Automatic"   -ErrorAction Stop
Position       : At line:1 char:1
                 + New-AzureRmVmss  -ResourceGroupName parytest1   -Location "East US"   ...
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
StackTrace     :    at Microsoft.Azure.Commands.Common.Strategies.SyncTaskScheduler.Wait(Task task, Action
                 progressUpdate)
                    at Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmss.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
HistoryId      : 16

```

**Comment by praries880** [comment]
@DanielHWe

#### Issue description
The [simple param set](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Compute/Commands.Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs#L257) for New-AzureRmVmss is meant to be a one shot command that creates all the required dependencies. 

LoadBalancer being a critical component of a VMSS, the cmdlet has tight built in requirements for it and tries to configure on accordingly. The behavior seen here is a bug. The cmdlet should have errored out with a message communicating that the LoadBalancer needs to reconfigured to be used with the cmdlet. We will be tracking the fix for that in [this issue](https://github.com/Azure/azure-powershell/issues/6473).

Here is a link to a [documentation issue](https://github.com/Azure/azure-powershell/issues/6474) which will be used to track generation of documentation that lists examples of the usages of the simple parameter set in New-AzureRmVmss.

We currently have an [issue](https://github.com/Azure/azure-powershell/issues/5659) open to add a _Probe_ parameter to configure a Probe on the scale set.

#### Workaround/Suggested way of configuring a Probe on the Load balancer
Create the vmss with the cmd :
```powershell
new-azurermvmss -Credential $vmCred -VMScaleSetName $scaleSetName
```
You can specify a custom name for the loadbalancer if you want to (by passing int the `LoadBalancerName`)

The cmd (if no LoadBalancerName is specified) will create a LB with the name `$scaleSetName`

Here's the steps you need to workaround as of now to configure a Probe on the LB for the Vmss :

```powershell
NewAzureRmVmss -VMScaleSetName $scaleSetName -Credential $vmCred -ResourceGroupName $rgName
#the cmdlet above will create the LB with the name $scaleSetName

#get the LB config
$lb=Get-AzureRmLoadBalancer -ResourceGroupName $rgName -Name $scaleSetName

#get the LB rule name
$lbRuleName=$lb.LoadBalancingRules[0].Name

#create the Probe config
$probe = New-AzureRmLoadBalancerProbeConfig -Name "MyProbe" -Protocol "http" -Port 80 -IntervalInSeconds 15 -ProbeCount 2 -RequestPath "healthcheck.aspx"

 #Get the lb rule config
$lbRule = Get-AzureRmLoadBalancerRuleConfig -Name $lbRuleName -LoadBalancer $lb

#configure the probe on the LB and the LBRule
$lbrule.Probe = $probe
$lb.Probes=$probe

 #Make any more LB configuration changes here

 #Save the LB config
Set-AzureRmLoadBalancer -LoadBalancer $lb

 #Done.. The probe is now configured on the LB/LB Rule
```

Do let us know if this helps you get unblocked.

#### Feedback requested
It would be great if you could provide us with the following information as well :
* We would also like to know what your typical VM/VMSS creation scenarios that are currenlty not covered by the simple param sets in both New-AzureRmVm and NewAzureRmVmss cmdlets

* Would it suffice if we add the follwing parameters `-Protocol "http" -Port 80 -IntervalInSeconds 15 -ProbeCount 2 -RequestPath "healthcheck.aspx` to the New-AzureRmVmss simple param set for configuring the probe (where the name of the Probe is `$scaleSetName-Probe`) or would you like to customize the probe name as well?

**Comment by DanielHWe** [comment]
For me it would works also to use the workaround. 

My use case is to create the VM scale set and the LoadBlancer for using it with an Application gateway. The instance VMs would serve http and https, so I have to add more than one health check. So parameter would be nice but not fully match.

In general I want to do a Blue/green deployment like described under [http://work.haufegroup.io/Blue-Green-Deployment-on-Azure/](http://work.haufegroup.io/Blue-Green-Deployment-on-Azure/) but using Costum Script Extensions instead of images.

So I have also trouble with using an existing vnet. I also tried by:

```
$vmss = New-AzureRmVmssConfig -Location $location -SkuCapacity 2 -SkuName "Standard_DS1_v2" -UpgradePolicyMode "manual" -ErrorAction Stop

Add-AzureRmVmssNetworkInterfaceConfiguration -VirtualMachineScaleSet $vmss -Name "vmssNetwork" -Primary $true -IPConfiguration $ipConfig
Set-AzureRmVmssStorageProfile -OsDiskOsType Windows -VirtualMachineScaleSet $vmss 
Set-AzureRmVmssOSProfile -ComputerNamePrefix $vmNamePrefix -AdminUsername $adminUsername -AdminPassword $adminPassword -VirtualMachineScaleSet $vmss

New-AzureRmVmss -ResourceGroupName $currentrg -Name $vmssName -VirtualMachineScaleSet $vmss -Verbose -ErrorAction Stop;
```
I get the error:

```
New-AzureRmVmss : 'CreateOption' cannot be null.
```

But with the error message I do not know what went wrong. Where is the CreateOption set that is missing?

**Comment by kasunsjc** [comment]
Same for me 
Following this guide at azure docs 
https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ssl-powershell

C:\> New-AzureRmVmss `
>>   -ResourceGroupName myResourceGroupAG `
>>   -Name myvmss `
>>   -VirtualMachineScaleSet $vmssConfig `
>>   -Verbose
VERBOSE: Performing the operation "New" on target "myvmss".
New-AzureRmVmss : 'CreateOption' cannot be null.
At line:1 char:1
+ New-AzureRmVmss `
+ ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzureRmVmss], ValidationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmss

**Comment by kasunsjc** [comment]
Need to add the -OsDiskCreateOption FromImage parameter its missing in the docs
C:\> Set-AzureRmVmssStorageProfile $vmssConfig `
>>   -ImageReferencePublisher MicrosoftWindowsServer `
>>   -ImageReferenceOffer WindowsServer `
>>   -ImageReferenceSku 2016-Datacenter `
>>   -ImageReferenceVersion latest `
>>   **-OsDiskCreateOption FromImage**

**Comment by praries880** [comment]
@DanielHWe Thanks for your feedback WRT the Probe param.

@DanielHWe @kasunsjc I will look into the issues you are pointing out WRT creating the vmss according to the linked sites.

**Comment by kasunsjc** [comment]
@praries880 I already create Issue for relevant doc refer below 
https://github.com/MicrosoftDocs/azure-docs/issues/11037

**Comment by praries880** [comment]
Thanks @kasunsjc

**Comment by praries880** [comment]
@hyonholee 
According to the validate command for the [OSDisk model](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/Generated/Models/VirtualMachineScaleSetOSDisk.cs#L160), `CreateOption` is a required property.

Following instructions [here](https://docs.microsoft.com/en-us/azure/application-gateway/tutorial-ssl-powershell#create-a-virtual-machine-scale-set) when i do create the StorageProfile, I dont see the property `CreateOption` set on the OSDisk : 
```powershell

PS D:\Public\azure-powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Compute> Set-AzureRmVmssSto
rageProfile $vmssConfig `
>>   -ImageReferencePublisher MicrosoftWindowsServer `
>>   -ImageReferenceOffer WindowsServer `
>>   -ImageReferenceSku 2016-Datacenter `
>>   -ImageReferenceVersion latest


Sku                           :
  Name                        : Standard_DS2
  Capacity                    : 2
UpgradePolicy                 :
  Mode                        : Automatic
  AutomaticOSUpgrade          : False
VirtualMachineProfile         :
  StorageProfile              :
    ImageReference            :
      Publisher               : MicrosoftWindowsServer
      Offer                   : WindowsServer
      Sku                     : 2016-Datacenter
      Version                 : latest
    OsDisk                    :
      WriteAcceleratorEnabled : False
Location                      : westus
```

The cmdlet [`Set-AzureRmVmssStorageProfile`](https://github.com/Azure/azure-powershell/blob/f34c546fea58d48819e968cab9ac3dd5c50551f3/src/ResourceManager/Compute/Stack/Commands.Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs#L230) does not require the [OSDiskCreateOption](https://github.com/Azure/azure-powershell/blob/f34c546fea58d48819e968cab9ac3dd5c50551f3/src/ResourceManager/Compute/Stack/Commands.Compute/Generated/VirtualMachineScaleSet/Config/SetAzureRmVmssStorageProfileCommand.cs#L82) param. This needs to be fixed. 

We need to 
* either pick a value that makes sense (I think `FromImage`) is appropriate in this case, or 
* we need to make the param 'OSDiskCreateOption` mandatory, this becomes a breaking change though. 

I would preffer option one here.

**Comment by DanielHWe** [comment]
In case I add "-OSDiskCreateOption FromImage" in my sample  so I use 

````
Set-AzureRmVmssStorageProfile -OsDiskOsType Windows -VirtualMachineScaleSet $vmss  -OSDiskCreateOption FromImage
````
 give me the a bit ambiguous error message:

```
New-AzureRmVmss : Cannot specify user image overrides for a disk already defined in the specified image reference.
ErrorCode: InvalidParameter
ErrorMessage: Cannot specify user image overrides for a disk already defined in the specified image reference.
StatusCode: 400
ReasonPhrase: Bad Request
OperationID : ed6d3f52-6eaa-4297-aae9-cdb18ae5de19
In D:\ExportedTemplate-TopasInfra-T-RG\azure_create_bg_loadBalancer.ps1:344 Zeichen:1
+ New-AzureRmVmss -ResourceGroupName $currentrg -Name $vmssName -Virtua ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzureRmVmss], ComputeCloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmss
````

But @kasunsjc gave me the current hint and it works with 

````
Set-AzureRmVmssStorageProfile -VirtualMachineScaleSet $vmss -ImageReferencePublisher MicrosoftWindowsServer -ImageReferenceOffer WindowsServer -ImageReferenceSku 2016-Datacenter -ImageReferenceVersion latest -OsDiskCreateOption FromImage
````

**Comment by kasunsjc** [comment]
@DanielHWe are you trying to update the existing VMSS ?

**Comment by DanielHWe** [comment]
No, it is during the creation of a new VMSS.

Full not working script is (giving Cannot specify user image overrides for a disk already defined):
```
$vmss = New-AzureRmVmssConfig -Location $location -SkuCapacity 2 -SkuName "Standard_DS1_v2" -UpgradePolicyMode "manual" -ErrorAction Stop

Add-AzureRmVmssNetworkInterfaceConfiguration -VirtualMachineScaleSet $vmss -Name "vmssNetwork" -Primary $true -IPConfiguration $ipConfig
Set-AzureRmVmssStorageProfile -OsDiskOsType Windows -VirtualMachineScaleSet $vmss  -OSDiskCreateOption FromImage
Set-AzureRmVmssOSProfile -ComputerNamePrefix $vmNamePrefix -AdminUsername $adminUsername -AdminPassword $adminPassword -VirtualMachineScaleSet $vmss

New-AzureRmVmss -ResourceGroupName $currentrg -Name $vmssName -VirtualMachineScaleSet $vmss -Verbose -ErrorAction Stop;

```

Working version:
```
$vmss = New-AzureRmVmssConfig -Location $location -SkuCapacity 2 -SkuName "Standard_DS1_v2" -UpgradePolicyMode "manual" -ErrorAction Stop

Add-AzureRmVmssNetworkInterfaceConfiguration -VirtualMachineScaleSet $vmss -Name "vmssNetwork" -Primary $true -IPConfiguration $ipConfig
Set-AzureRmVmssStorageProfile -VirtualMachineScaleSet $vmss -ImageReferencePublisher MicrosoftWindowsServer -ImageReferenceOffer WindowsServer -ImageReferenceSku 2016-Datacenter -ImageReferenceVersion latest -OsDiskCreateOption FromImage
Set-AzureRmVmssOSProfile -ComputerNamePrefix $vmNamePrefix -AdminUsername $adminUsername -AdminPassword $adminPassword -VirtualMachineScaleSet $vmss

New-AzureRmVmss -ResourceGroupName $currentrg -Name $vmssName -VirtualMachineScaleSet $vmss -Verbose -ErrorAction Stop;
```

**Comment by kasunsjc** [comment]
@DanielHWe In Not working version its missing few parameters 

Set-AzureRmVmssStorageProfile -OsDiskOsType Windows -VirtualMachineScaleSet $vmss  -OSDiskCreateOption FromImage

Set-AzureRmVmssStorageProfile -VirtualMachineScaleSet $vmss -ImageReferencePublisher MicrosoftWindowsServer -ImageReferenceOffer WindowsServer -ImageReferenceSku 2016-Datacenter -ImageReferenceVersion latest -OsDiskCreateOption FromImage

If you compare above two not working command misses 
-ImageReferencePublisher MicrosoftWindowsServer -ImageReferenceOffer WindowsServer -ImageReferenceSku 2016-Datacenter -ImageReferenceVersion latest parameters

**Comment by DanielHWe** [comment]
Yes some parameters are missing, so I would expect an error saying parameters are missing not an error message say "Cannot specify user image overrides for a disk already defined in the specified image reference.".

With that error message I was thinking there is a wrong parameter not a missing one.

**Comment by praries880** [comment]
@DanielHWe , @kasunsjc  Thanks for the input here.
@hyonholee ,  @huangpf Can you guys look at the error message reported by Daniel above and comment on it?

**Comment by teresaritorto** [comment]
This issue has been marked as stale and is being closed due to inactivity. If you are still experiencing problems, please create a new issue and reference the latest version of the Azure PowerShell tool. This helps us ensure that we address current scenarios and provide accurate support. Thank you for your understanding!
